server/src/main/java/io/crate/types/NumericType.java

```diff
@@ -163,6 +163,12 @@ public BigDecimal sanitizeValue(Object value) {
             BigInteger bigInt = BigInteger.valueOf(longValue);
             return new BigDecimal(bigInt, scale == null ? 0 : scale, mathContext());
         }
+        // Can be double if coming from source
+        if (value instanceof Double doubleVal) {
```

Comment:
Question on code comment: don't we store BigDecimal in the source as is?
Referring to Indexer -> TranslogIndexer -> writeValue.